tensorflow/core/util/cuda_kernel_helper.h

```diff
@@ -23,10 +23,82 @@ limitations under the License.
 #include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
 #include "tensorflow/core/platform/types.h"
 
+// Usage of GetCudaLaunchConfig, GetCuda2DLaunchConfig, and GetCuda3DLaunchConfig:
+//
+// There are two versions of GetCudaLaunchConfig and GetCuda2DLaunchConfig, one
```

Comment:
Yes, I agree, but there is another thing to consider: The 1D maximum occupancy version is introduced in https://github.com/tensorflow/tensorflow/commit/3c02d1100788789b04e04feb93761f0ad898ea77 and given the same symbol `GetCudaLaunchConfig` half a month ago. I'm not sure how to deal with this. Three options are:

- Keep `GetCudaLaunchConfig` unchanged, and rename the maximize occupancy version of 2D and 3D to `Get2DCudaMaxOccupancyLaunchConfig` and `...3D...`. This might make things even more confusing.
- Rename all these functions, and change the kernel launches affected. This makes things clearer, but might break somebody's code. Are functions in this file considered API that should avoid change?
- Leave these functions as it is (i.e. overload the same symbol)

Which one do you prefer?